Adyen/Components/Await/AwaitComponent.swift

```diff
@@ -81,7 +81,7 @@ public final class AwaitComponent: AnyAwaitComponent {
             let scheduler = BackoffScheduler(queue: .main)
             let baseAPIClient = APIClient(environment: environment)
```

Comment:
What about injecting `apiClient` into AwaitComponent? Why does it have to know how to generate one?
Maybe we can use a Builder pattern here, so AwaitComponent does not have to know about dependencies of any underling component?

Reply:
I don't want to enable client code to inject the `apiClient`, for the `AwaitComponent` to work as expected the apiClient HAS to be a `RetryAPICliet` with a `BackoffScheduler`, `RetryAPICliet` with a `BackoffScheduler` is an integral part of how `AwaitComponent` work, and I enabled injecting the `apiClient` in an internal initializer for `Unit tests` only :-) .

Reply:
please check the last commit, I Renamed `AnyAwaitComponent` into `AnyAwaitActionHandler`, moved the payment method specific `AwaitActionHandler` building into `AwaitActionHandlerProvider` builder and modified unit tests AwaitComponentTests